### DISCUSSION TITLE
[@types/express] Should express locals not force extension of Record<string, any>?

### DISCUSSION THREAD

**Discussion by djMax** [DefinitelyTyped/DefinitelyTyped#62776] [Issues with a @types package]
I'm rebuilding a large bit of microservice infrastructure around Typescript and Express 5. In general it's going well, BUT, because the Express Request/App/Response types force the generic type arguments for locals to extend from Record<string, any>, I can't get proper type checking on those locals. If you misspell a property name, TS doesn't care. Should we just remove the type constraint there? Or should it extend {} instead? For some reason, when I pass in a Locals type that does NOT extend, everything seems to work properly, but I don't really get why...

[a comment by an automated account is not shown]

**Comment by chrisb2244** [comment]
I'm having the same problem - is there a workaround or is this impossible without modifying `@types/express-serve-static-core/index.d.ts`? I couldn't quite work out from your description if that was what you were doing (locally modifying the node_modules definitions from this package) or if you'd managed to put something in a separate type declaration file.
